### PR TITLE
Fix layers visualisation for transcoded dataset

### LINKED ISSUE #614: Expand and collapse modular pipeline structure doesn't work with transcoding datasets
## Description

`kedro viz` fails to launch if a modular pipeline contains a transcoded dataset.

## Context
How has this bug affected you? What were you trying to accomplish?

## Steps to Reproduce

Launch `kedro viz` in a project with a modular pipeline containing transcoded dataset. Observe that it throws an exception as follow:

```
Traceback (most recent call last):
  File "/Users/lim_Hoang/opt/anaconda3/envs/kedro-viz38/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/Users/lim_Hoang/opt/anaconda3/envs/kedro-viz38/lib/python3.8/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/lim_Hoang/Projects/kedro-viz/package/kedro_viz/server.py", line 83, in run_server
    populate_data(data_access_manager, catalog, pipelines, session_store_location)
  File "/Users/lim_Hoang/Projects/kedro-viz/package/kedro_viz/server.py", line 42, in populate_data
    data_access_manager.add_pipelines(pipelines)
  File "/Users/lim_Hoang/Projects/kedro-viz/package/kedro_viz/data_access/managers.py", line 85, in add_pipelines
    self.add_pipeline(registered_pipeline_id, pipeline)
  File "/Users/lim_Hoang/Projects/kedro-viz/package/kedro_viz/data_access/managers.py", line 151, in add_pipeline
    self.modular_pipelines.add_output(
  File "/Users/lim_Hoang/Projects/kedro-viz/package/kedro_viz/data_access/repositories.py", line 331, in add_output
    raise ValueError(
ValueError: Attempt to add a non-data node as input to modular pipeline transcoding
```

## Expected Result

Kedro Viz should launch successfully.

## Actual Result

An exception was thrown.

## Your Environment
Include as many relevant details as possible about the environment you experienced the bug in:

* Web browser system and version:
* Operating system and version:
* NodeJS version used (if relevant):
* Kedro version used (if relevant):
* Python version used (if relevant):

## Checklist
- [ ] Include labels so that we can categorise your issue

### PR DESCRIPTION
**Pull request by limdauto**
## Description

Fixes #614 and #615

## QA notes

I have both tested manually and added unit tests for the fix.

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [x] Added tests to cover my changes

## Legal notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

### CODE REVIEW

File: package/tests/test_models/test_graph/test_graph_nodes.py [hunk 1]
```diff
@@ -216,7 +216,7 @@ def test_add_node_to_pipeline(self):
         another_pipeline = RegisteredPipeline("testing")
         kedro_dataset = CSVDataSet(filepath="foo.csv")
         data_node = GraphNode.create_data_node(
-            full_name="dataset",
+            full_name="dataset@transcoded",
```
Comment (limdauto) [inline package/tests/test_models/test_graph/test_graph_nodes.py:219]: This covers #614

File: package/tests/test_data_access/test_repositories.py [hunk 2]
```diff
@@ -138,3 +140,21 @@ def test_add_output_should_raise_if_adding_non_data_node(self):
...
+                "filepath": "cars.csv",
+                "layer": "raw",
+            },
+            "cars@spark": {"type": "spark.SparkDataSet", "filepath": "cars.parquet"},
+        }
+        catalog = DataCatalog.from_config(catalog_config)
+        repo.set_catalog(catalog)
+        assert repo.get_layer_for_dataset("cars") == "raw"
+        assert repo.get_layer_for_dataset("cars@pandas") == "raw"
+        assert repo.get_layer_for_dataset("cars@spark") == "raw"
```
Comment (limdauto) [inline package/tests/test_data_access/test_repositories.py:160]: This covers #615

**Review by AhdraMeraliQB — APPROVED** [review]
LGTM!

**Review by rashidakanchwala — APPROVED** [review]
LGTM - i was able to test 615 and the layers show up properly! thanks

**Pull request merged**